src/sidebar/helpers/export-annotations.ts

```diff
@@ -0,0 +1,50 @@
+import type { DocumentMetadata } from '../../types/annotator';
+import { truncate } from '../util/unicode';
+
+/** Format a date as an ISO `YYYY-MM-DD` string. */
+export function formatDate(date: Date): string {
+  const year = date.getFullYear();
+  const month = `${date.getMonth() + 1}`.padStart(2, '0');
+  const day = `${date.getDate()}`.padStart(2, '0');
+
+  return `${year}-${month}-${day}`;
+}
+
+export type SuggestedFilenameOptions = {
+  /** Metadata of the document the annotations are associated with. */
+  documentMetadata?: DocumentMetadata;
+
+  /** Name of the group containing the exported annotations. */
+  groupName?: string;
+
+  /** Date of the export. Defaults to the current date. */
+  date?: Date;
+};
+
+/**
+ * Generate a default filename for a file containing an exported set of
+ * annotations.
+ *
+ * The generated filename may not meet the character, length or other
+ * restrictions of the user's system. It is assumed the browser will modify the
+ * filename as needed when it downloads the file.
+ */
+export function suggestedFilename({
+  documentMetadata,
+  groupName,
+  date = new Date(),
+}: SuggestedFilenameOptions) {
+  const filenameSegments = [formatDate(date)];
+
+  if (documentMetadata?.title) {
+    filenameSegments.push(truncate(documentMetadata.title, 50));
```

Comment:
Not super critical, but should we replace spaces with dashes in the title, as we do with the group name?

Reply:
I'm not sure. Filenames in general can have spaces in them. We can ask around in Slack and see what folks think.

Reply:
To be honest, I made the decision of replacing spaces in the group name pretty arbitrarily, so I'm perfectly fine with this.

Perhaps we should also keep them in the group name.